### PR TITLE
Added logging for changed overlayfs dirs

### PR DESCRIPTION
**Pull request by chofnar**
This PR implements the suggestions from issue #159.
Without --debug, it will print the first 2 changed dirs:
```
cache miss because content of 3 overlay dirs changed:
/home/chofnar/stacker/f2
/home/chofnar/stacker/f3
and 1 others. use --debug for complete output
```

With --debug, it will print all modifications:

```
cache miss because content of 3 overlay dirs changed:
/home/chofnar/stacker/f2
/home/chofnar/stacker/f3
/home/chofnar/stacker/f4
```

Newest change also lists the files that changed:
With --debug
```
cache miss because content of 4 overlay dirs changed:
Changed overlay_dir: /home/piggie/stackeryamls/od1
Contents that changed:
/home/piggie/stackeryamls/od1/subdir/subdirfile.txt
/home/piggie/stackeryamls/od1/file1.txt
Changed overlay_dir: /home/piggie/stackeryamls/od2
Contents that changed:
/home/piggie/stackeryamls/od2/file2.txt
Changed overlay_dir: /home/piggie/stackeryamls/od3
Contents that changed:
/home/piggie/stackeryamls/od3/file3.txt
Changed overlay_dir: /home/piggie/stackeryamls/od4
Contents that changed:
/home/piggie/stackeryamls/od4/file4.txt
```

Without --debug
```
cache miss because content of 4 overlay dirs changed:
Changed overlay_dir: /home/piggie/stackeryamls/od1
Contents that changed:
/home/piggie/stackeryamls/od1/file1.txt
/home/piggie/stackeryamls/od1/subdir/subdirfile.txt
Changed overlay_dir: /home/piggie/stackeryamls/od2
Contents that changed:
/home/piggie/stackeryamls/od2/file2.txt
and 2 other overlay_dirs. use --debug for complete output
```
Signed-off-by: Catalin Hofnar <catalin.hofnar@gmail.com>

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -272,11 +273,27 @@ func (c *BuildCache) Lookup(name string) (*CacheEntry, bool, error) {
...
+	if overlayDirsChanged {
+		log.Infof("cache miss because content of %d overlay dirs changed:", len(changedOverlayDirs))
+		if c.config.Debug {
+			for _, dir := range changedOverlayDirs {
+				log.Infof(dir)
+			}
+		} else {
+			for index, dir := range changedOverlayDirs {
+				log.Infof(dir)
+				if index == 1 && len(changedOverlayDirs) > 2 {
```
Comment (rchincha) [inline cache.go:289]: Move the "if c.config.Debug" check here, since you are doing log.Infof(dir) in both cases

Comment (chofnar) [reply cache.go]: Modified as requested

**Comment by peusebiu** [comment]
LGTM

**Comment by peusebiu** [comment]
LGTM

**Review by andaaron — APPROVED** [review]
Change is good, but I think the challenge was to also show what content of the overlay dir was changed (except we don't store that kind of information as far as I see). @raharper was that the ask?

Maybe it would make sense to enhance the caching logic to cache the hash per file instead of the entire folder?

**Comment by chofnar** [comment]
> Change is good, but I think the challenge was to also show what content of the overlay dir was changed (except we don't store that kind of information as far as I see). @raharper was that the ask?
> 
> Maybe it would make sense to enhance the caching logic to cache the hash per file instead of the entire folder?

The logic already supported something like that, with some small adaptations. Please check the new description for the updated output

**Review by andaaron — APPROVED** [review]
Looks good.
Though I think we don't really need the log messages:
```
Changed overlay_dir: <folder_path>
Contents that changed:`
```
 The user can figure it out on his own based on the files shown.

**Review by hallyn — APPROVED** [review]

**Pull request merged**